src/Playwright/Core/BrowserContext.cs

```diff
@@ -166,6 +168,11 @@ public async Task CloseAsync()
         {
             try
             {
+                if (IsClosed)
```

Comment:
Rename this to IsClosedOrClosing, set it from Channel_Closed handler.

Reply:
Done